src/components/views/elements/PowerSelector.tsx

```diff
@@ -106,27 +111,37 @@ export default class PowerSelector<K extends undefined | string> extends React.C
         });
     }
 
-    private onSelectChange = (event: React.ChangeEvent<HTMLSelectElement>): void => {
+    private onSelectChange = async (event: React.ChangeEvent<HTMLSelectElement>): Promise<void> => {
         const isCustom = event.target.value === CUSTOM_VALUE;
         if (isCustom) {
             this.setState({ custom: true });
         } else {
             const powerLevel = parseInt(event.target.value);
-            this.props.onChange(powerLevel, this.props.powerLevelKey);
             this.setState({ selectValue: powerLevel });
+            try {
+                await this.props.onChange(powerLevel, this.props.powerLevelKey);
+            } catch {
+                if (this.unmounted) return;
+                this.initStateFromProps();
```

Comment:
comment here would be nice

```suggestion
                // If the request failed, roll back the state of the selector.
                this.initStateFromProps();
```